src/NLog/Internal/TransformedLayout.cs

```diff
@@ -0,0 +1,126 @@
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+using System;
+using JetBrains.Annotations;
+using NLog.Config;
+using NLog.Layouts;
+
+namespace NLog.Internal
+{
+    /// <summary>
+    /// Transformation on a layout. Pre-calculated if the layout has a fixed value.
+    /// </summary>
+    internal class TransformedLayout : IRenderable, ISupportsInitialize
+    {
+        private readonly string _fixedValue;
+        [CanBeNull] private readonly Func<Layout, LogEventInfo, string> _renderLogEvent;
+        private readonly Func<string, string> _transformation;
+
+        /// <summary>
+        /// Create TransformedLayout.
+        /// </summary>
+        /// <param name="layout">Layout</param>
+        /// <param name="transformation">Transformation method</param>
+        /// <param name="renderLogEvent">Optional renderer, otherwise <see cref="Layouts.Layout.Render" /> is used</param>
+        private TransformedLayout([NotNull] Layout layout, [NotNull] Func<string, string> transformation, [CanBeNull] Func<Layout, LogEventInfo, string> renderLogEvent)
+        {
+            _transformation = transformation ?? throw new ArgumentNullException(nameof(transformation));
+            _renderLogEvent = renderLogEvent;
+            Layout = layout ?? throw new ArgumentNullException(nameof(layout));
+
+            if (layout is SimpleLayout s && s.IsFixedText)
+            {
+                _fixedValue = transformation(s.FixedText);
+            }
+        }
+
+        public Layout Layout { get; }
+
+        #region Implementation of IRenderable
+
+        /// <inheritdoc />
+        public string Render(LogEventInfo logEvent)
+        {
+            if (_fixedValue != null)
+            {
+                return _fixedValue;
+            }
+
+            var value = _renderLogEvent != null ? _renderLogEvent(Layout, logEvent) : Layout.Render(logEvent);
+
+            return _transformation(value);
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Create TransformedLayout. If <paramref name="layout" /> is null, then the returns null
+        /// </summary>
+        /// <param name="layout">Layout</param>
+        /// <param name="transformation">Transformation method</param>
+        /// <param name="renderLogEvent">Optional renderer, otherwise <see cref="Layouts.Layout.Render" /> is used</param>
+        /// <returns>null if <paramref name="layout" /> is null</returns>
+        public static TransformedLayout Create([CanBeNull] Layout layout,
```

Comment:
FYI passing the `renderLogEvent` is a bit pity, but guess it's needed.

Reply:
I'm guessing that is the whole idea with this class to wrap a Layout with a post-processing-method (renderLogEvent). So not sure why it is a pity.

Reply:
because we have also `Layout.Render` and not also `renderLogEvent` . So 2 methods for the same thing (but with other caching)